apps/backend/src/routes/sendEmail.js

```diff
@@ -0,0 +1,42 @@
+const express = require('express');
+const { getTransporter } = require('../utils/getTransporter');
+const { withEmailTemplate } = require('../utils/withEmailTemplate');
+
+const router = express.Router();
+
+router.post('/', async (req, res) => {
+  try {
+    const emailDataArray = req.body;
+
+    const emailInfoArray = [];
+
+    for (const emailData of emailDataArray) {
+      const { from, to, subject, content, isPersonalEmail, replyTo } = emailData;
+
+      const transporter = await getTransporter(isPersonalEmail);
+
+      if (!transporter) {
+        continue;
+      }
+
+      const info = await transporter.sendMail({
+        from: from || '"Best from ClassroomIO" <best@classroomio.com>',
+        to,
+        subject,
+        replyTo,
+        html: withEmailTemplate(content)
+      });
+
+      console.log('Message sent: %s', info.messageId);
+      emailInfoArray.push(info);
+    }
+
+    console.log('sendmail');
+    return res.json({ info: emailInfoArray });
```

Comment:
Please update this code to:

```ts
Promise.all(emailDataArray.map(async emailData => {
  try {
  const { from, to, subject, content, isPersonalEmail, replyTo } = emailData;
  // write the email logic here
  } catch (error) {
    console.error('Error sending email', error);
  }
});

return res.json({ success: true });
```

The difference between this approach is that we want to return success early in the code so that the frontend doesn't have to wait for all the emails to send.

Meanwhile the emails will be sending on the server

Reply:
if we return success early and for any reason there's an issue while sending one of the mails,how do we handle that?

Reply:
For now nothing.

Ideally we should be using a queuing system that can retry sending if it fails.

For now we will go with this and monitor it.